src/Core/ServerSettings.h

```diff
@@ -81,7 +81,7 @@ namespace DB
     M(UInt64, mmap_cache_size, DEFAULT_MMAP_CACHE_MAX_SIZE, "A cache for mmapped files.", 0) \
     \
     M(Bool,   disable_internal_dns_cache, false, "Disable internal DNS caching at all.", 0) \
-    M(UInt64, dns_cache_max_size, 1024, "Internal DNS cache max size in bytes.", 0) \
+    M(UInt64, dns_cache_max_size, (1024 * 1024), "Internal DNS cache max size in bytes.", 0) \
```

Comment:
We should `#include <base/unit.h>` and use more readable quantities like `1_MB`.